ramalama/model.py

```diff
@@ -389,7 +401,15 @@ def get_gpu():
             content = file.read()
             if "asahi" in content.lower():
                 return "ASAHI_VISIBLE_DEVICES", 1
-
+                
+    try:
+        #TODO I don't currently have access to a PC w/ multiple NVidia GPUs nor an NVidia Mac...  but I *think* that
+        #every Linux and Windows machine having modern NVidia will have nvidia-smi and that the number of lines corresponds to the number of zero-indexed gpus
+        check_output = subprocess.run(['nvidia-smi', '-L'], check=True, capture_output=True) #shell to nvidia-smi
```

Comment:
Use run_cmd

Reply:
We can go a step further and query the nvidia-smi command itself to get more info! For example doing this command `nvidia-smi --query-gpu=index,memory.total --format=csv,noheader,nounits | sort -t, -k2 -nr | head -n 1` Can get us the largest vram GPU and id formatted as "id, vram-in-mb".

We can do something like this

```
try:
        check_output = subprocess.run(['nvidia-smi', '-L'], check=True, capture_output=True) #shell to nvidia-smi
 if not check_output.returncode: #if command EXIT_SUCCESS
            return "CUDA_VISIBLE_DEVICES", len(check_output.stdout.splitlines()) # ret cuda, #gpus?
    except Exception: {} #fall through

    return None, None
```

Reply:
I think that's a fabulous idea.  My inclination in general is to avoid shelling out, but I don't think we're going to find a better or more lightweight way to test for the presence of NVidia and CUDA.  Probably why all the NVidia Container Toolkit docs seem to use it as a sanity check for installs.

Reply:
Same I'd rather not shell out myself but doing it this way avoids complications with different systems. if we can assume a system has an nvidia GPU then most likely there will be drivers installed with nvidia-smi as well. 

Right now the vulkan backed for llama.cpp doesn't have all the functionality like cuda and hip blas does. But later down the line id like to switch to vulkan and use the vulkan SDK to query GPU data since it supports amd nvidia and intel graphics.